### DISCUSSION TITLE
Unable to compile Zed on void linux

### DISCUSSION THREAD

**Discussion by lcassa** [zed-industries/zed#10990] [Feature Requests]
I'm getting this error trying to build zed on Void Linux

``` 
= note: /usr/sbin/ld: cannot find -lasound: No such file or directory

          /usr/sbin/ld: cannot find -lxkbcommon: No such file or directory
          /usr/sbin/ld: cannot find -lxkbcommon-x11: No such file or directory
          collect2: error: ld returned 1 exit status
```
I already have libxkb dependencies installed
``` xbps-query -s *xkb*            
[*] libxkbcommon-1.7.0_1       Library to handle keyboard descriptions
[*] libxkbcommon-devel-1.7.0_1 Library to handle keyboard descriptions - development files
[*] libxkbcommon-x11-1.7.0_1   Library to handle keyboard descriptions - X11 support
[*] libxkbfile-1.1.3_1         Xkbfile Library from X.org
[*] libxkbregistry-1.7.0_1     XKB API to query available rules, models, layouts, variants...
[*] xkbcomp-1.4.7_1            XKBD keymap compiler
``` 
and alsa as well
``` xbps-query -s *alsa*
[*] alsa-lib-1.2.11_1       Advanced Linux Sound Architecture (ALSA) library
[*] alsa-lib-devel-1.2.11_1 Advanced Linux Sound Architecture (ALSA) library - development...
``` 
Can anybody help me out?

**Comment by aohanhongzhi** [comment]
have you see the document https://github.com/zed-industries/zed/blob/main/docs/src/development/linux.md

and run 
```shell
script/linux
```
to Install the necessary system libraries.

**Reply by DarkCat09** [reply]
Ran the script, still `-lasound`, `-lxkbcommon` and `-lxkbcommon-x11` are not found. Latest Rust 1.80, Void Linux musl x86_64.

**Comment by rudolf-enum** [comment]
It's because of how musl works with dynamic libraries (it doesn't), so there's an extra flag needed. Had you used Void with glibc, it would've worked.

I've added this snippet right above `cargo build ...` in `./script/bundle-linux` and it works now on my Alpine Linux machine, I'll try to make a PR.

```sh
if echo "$target_triple" | grep musl; then
  export RUSTFLAGS="$RUSTFLAGS -C target-feature=-crt-static"
fi
```

**Comment by Strix-Vyxlor** [comment]
It's still not working for me,
it tries adding aarch64-unknown-linux-musl-musl as a toolchain (compiling on a rpi5 using void musl).
I fixed this by removing the -musl from the musl_tripple variable in bundle-linux.
but it still craches with -lxkbcommon -lxkbcommon-x11 -lasound no such file or directory.
even after adding `RUSTFLAGS="-L /usr/lib -L /usr/lib/gcc/aarch64-linux-musl/13.2 -L /usr/lib64 -L /usr/lib64/gcc/aarch64-linux-musl/13.2 -C target-feature=-crt-static"` before ./scripts/bundle-linux.
Can anybody help?

I'm having a similar isue with cosmic, [see this here on reddit](https://www.reddit.com/r/voidlinux/comments/1hk7cdj/need_help_with_compiling_cosmiceposh/)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
